TESTING.md

```diff
@@ -1,89 +1,483 @@
+ * ❌ Installing a plugin through the admin panel (far too specific, not useful)
+ * ✅ Specifying a different UID using the `containerSecurityContext.runAsUser` in `values.yaml` and checking it (checks a feature intrinsic to the Chart)
+
+The tests may be regarded as _deployment_ tests since their goal is to verify that the software is correctly deployed with all the inherent features. Both functional and non-functional characteristics are evaluated in these tests, focusing on the installation aspect.
+
+### The strategy
+
+Before writing any test scenario, understand the primary purpose of the chart and its components. Take a look at [the documentation about the chart under test](https://github.com/bitnami/charts/tree/main/bitnami), explore the different templates and configurations in `values.yaml` and glance over the [docker image documentation](https://github.com/bitnami?q=docker&type=all&language=&sort=). This will give you a solid base for creating valuable test scenarios.
+
+As Charts are usually composed of a number of different components, it is also essential to test their integrations and the Chart as a whole. As a general guideline, testing a `bitnami/chart` can be reduced to:
+
+1. Identifying the components of the Chart and verifying their integration. *e.g. WordPress + MariaDB + PHP + Data Volume*
+1. Summarizing the main area features the asset offers and asserting the Chart delivers them. *e.g. Creating a post in a blog*
+1. Focusing on the unique features the Chart offers. *e.g. ConfigMaps, PVCs, Services, secrets, etc.*
+
+It is easily noticeable though that Charts are usually highly configurable artifacts. Through parameters exposed in `values.yaml`, it is fairly common to perform customizations that range from enabling simple features (e.g. exporting metrics to Prometheus) to complete changes in the architecture of the application that will be deployed (e.g. standalone vs. main-secondary replication in DBs). In order to cope with this high variability, we should:
+
+* Stick to the KISS (Keep It Short and Simple) principle: only test/consider the features that are enabled by default.
+* When params allow to customize the deployment architecture, give preference to: (1) the more representative blueprint and (2) the one provides more code-covering.
+* Change the values for parameters of features enabled by default. This allows to ensure they are correctly picked up by the Chart.
+
+#### Runtime parameters
+
+Helm allows to customize how a Chart is deployed [during its installation](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing). The subset of parameters from `values.yaml` (and their values) that are passed in [during a Helm Chart installation](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing) are referred to as runtime/installation/deployment parameters in this guide. They are of **uttermost** importance as they have a great impact on the instance under test.
```

Comment:
Is the duplicated use of the link done on purpose? I think we can leave only one of them.

Reply:
Yes, I think I missed removing one of the two.